pyramid_swagger/api.py

```diff
@@ -101,26 +101,206 @@ def view_for_api_declaration(request):
+        ('file://', 'file.'),
+        ('http://', 'http.'),
+        ('https://', 'https.'),
+    ]
+    # replacement over the whole string
+    replacement_patterns = [  # defined as tuples (un-marshaled, marshaled)
+        ('/', '..'),
+        ('#', '|'),
+    ]
+
+    # repl_idx allow to prevent if statements over the code execution
+    # instead of doing:
+    #       if is_marshaling:
+    #           x = x.replace(replacement[0], replacement[1])
+    #       else:
+    #           x = x.replace(replacement[1], replacement[0])
+    # we can achieve the same doing
+    #       x = x.replace(replacement[repl_idx], replacement[1-repl_idx])
+    repl_idx = 0 if is_marshaling else 1
+    replaced_scheme = ''
+    for replacement in scheme_replacements:
+        if path.startswith(replacement[repl_idx]):
+            replaced_scheme = replacement[1 - repl_idx]
+            path = path[len(replacement[repl_idx]):]
+            break
+    for replacement in replacement_patterns:
+        path = path.replace(replacement[repl_idx], replacement[1 - repl_idx])
+    return replaced_scheme + path
+
+
+def _get_absolute_link(spec, relative_path, current_path=''):
+    spec_url = urlparse(spec.origin_url)
+    spec_file = os.path.abspath(spec_url.path)
+    spec_dir = os.path.dirname(spec_file)
+
+    current_path_url = urlparse(current_path)
+
+    target_url = urlparse(relative_path)
+    if len(target_url.scheme) > 0:
+        target_scheme = target_url.scheme
+    elif len(current_path_url.scheme) > 0:
+        target_scheme = current_path_url.scheme
+    else:
+        target_scheme = 'file'
+
+    if target_scheme == 'file':  # targeting a local file
+        # if not absolute path, make it absolute
+        if not target_url.path.startswith('/'):
+            # if path is empty, then is targeting to the current file
+            filename = target_url.path if len(target_url.path) > 0 \
+                else os.path.basename(current_path_url.path)
+            target_url = urlparse('{path}#{fragment}'.format(
+                path=os.path.join(
+                    spec_dir,
+                    os.path.dirname(current_path_url.path),
+                    filename,
+                ),
+                fragment=target_url.fragment,
+            ))
+        return target_url
+
+    elif target_scheme in ['http', 'https']:
+        remote_path = os.path.abspath(os.path.join(
+            current_path_url.path if len(current_path_url.path) > 0 else '/',
+            target_url.path
+        ))
+        return urlparse('{scheme}://{hostname}{path}#{fragment}'.format(
+            scheme=target_scheme,
+            hostname=target_url.hostname,
+            path=remote_path,
+            fragment=target_url.fragment,
+        ))
+    else:
+        return urlparse(relative_path)
+
+
+def _relpath(path, start):
+    """Return a relative version of a path
+    NOTE: Code duplicated from Python 2.7.9 implementation because the
```

Comment:
I have found the [python issue #5117](http://bugs.python.org/issue5117) that seems explaining the difference in the ``os.path.relpath`` execution with Python 2.6.9, 2.7.9, 3.4.2 and 3.5.2.
A more direct approach could be to *practically* seen the problem. You can do it running the script that is shown below (are added also the results that I have on my laptop).

I have not found an *official* reply to your doubt; but I'm able to provide you a short snippet that highlights the issue.

Script
```bash
for version in 2.6.9 2.7.9 3.4.2 3.5.2; do
    pyenv global $version
    python -c "import os; import sys; print(sys.version); print(os.path.relpath('/dir1/../dir2/file.json', '/'))"
    echo "=========="
done
```

My Results
```bash
2.6.9 (unknown, Jul  1 2016, 10:58:34) 
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.31)]
../dir2/file.json
==========
2.7.9 (default, Jul  1 2016, 10:52:19) 
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.31)]
dir2/file.json
==========
3.4.2 (default, Jul  1 2016, 10:56:36) 
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.31)]
dir2/file.json
==========
3.5.2 (default, Jul  3 2016, 11:59:24) 
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.31)]
dir2/file.json
```
**NOTE**: I'm using pyenv to quickly switch between python version. Each command execution shows the used version and the result of the ``os.path.relpath('/dir1/../dir2/file.json', '/')`` command.